scipy/stats/unuran/setup.py

```diff
@@ -0,0 +1,161 @@
+        path = [str(dir_ / file) for file in files]
+        sources += [source for source in path if (source.endswith(".c"))]
+    return sources
+
+
+def _get_version(configure_dot_ac, target_name):
+    configure_dot_ac = pathlib.Path(__file__).parent / configure_dot_ac
+    with open(configure_dot_ac, "r") as f:
+        s = f.read()
+        start_idx = s.find(target_name)
+        end_idx = s[start_idx:].find(")") + len(s[:start_idx])
+        version = s[start_idx:end_idx].split(",")[1][1:-1]
+    return version
+
+
+def configuration(parent_package="", top_path=None):
+    import numpy as np
+    from numpy.distutils.misc_util import Configuration
+
+    config = Configuration("unuran", parent_package, top_path)
+
+    # UNU.RAN info
+    UNURAN_VERSION = _get_version("src/configure.ac", "AM_INIT_AUTOMAKE")
+    UNURAN_DIR = pathlib.Path(__file__).parent.resolve()
+
+    # Define macros assuming we have C99 standard headers
+    DEFINE_MACROS = [
+        ("HAVE_ALARM", "1"),
+        ("HAVE_DECL_ALARM", "1"),
+        ("HAVE_DECL_HUGE_VAL", "1"),
+        ("HAVE_DECL_INFINITY", "1"),
+        ("HAVE_DECL_ISFINITE", "1"),
+        ("HAVE_DECL_ISINF", "1"),
+        ("HAVE_DECL_ISNAN", "1"),
+        ("HAVE_DECL_LOG1P", "1"),
+        ("HAVE_DECL_SIGNAL", "1"),
+        ("HAVE_DECL_SNPRINTF", "1"),
+        ("HAVE_DECL_VSNPRINTF", "1"),
+        ("HAVE_FLOAT_H", "1"),
+        ("HAVE_FLOOR", "1"),
+        ("HAVE_IEEE_COMPARISONS", "1"),
+        ("HAVE_INTTYPES_H", "1"),
+        ("HAVE_LIBM", "1"),
+        ("HAVE_LIMITS_H", "1"),
+        ("HAVE_POW", "1"),
+        ("HAVE_SIGNAL", "1"),
+        ("HAVE_SQRT", "1"),
+        ("HAVE_STDINT_H", "1"),
+        ("HAVE_STDLIB_H", "1"),
+        ("HAVE_STRCHR", "1"),
+        ("HAVE_STRING_H", "1"),
+        ("HAVE_STRTOL", "1"),
+        ("HAVE_STRTOUL", "1"),
+        ("LT_OBJDIR", '".libs/"'),
+        ("PACKAGE", '"unuran"'),
+        ("PACKAGE_BUGREPORT", '"unuran@statmath.wu.ac.at"'),
+        ("PACKAGE_NAME", '"unuran"'),
+        ("PACKAGE_STRING", '"unuran %s"' % UNURAN_VERSION),
+        ("PACKAGE_TARNAME", '"unuran"'),
+        ("PACKAGE_URL", '""'),
+        ("PACKAGE_VERSION", '"%s"' % UNURAN_VERSION),
+        ("STDC_HEADERS", "1"),
+        ("UNUR_ENABLE_INFO", "1"),
+        ("VERSION", '"%s"' % UNURAN_VERSION),
+        ("HAVE_CONFIG_H", "1")
+    ]
+
+    UNURAN_DIRS = [
+        os.path.join("src"),
+        os.path.join("src", "src"),
+        os.path.join("src", "src", "distr"),
+        os.path.join("src", "src", "distributions"),
+        os.path.join("src", "src", "methods"),
+        os.path.join("src", "src", "parser"),
+        os.path.join("src", "src", "specfunct"),
+        os.path.join("src", "src", "urng"),
+        os.path.join("src", "src", "utils"),
+        os.path.join("src", "src", "tests"),
+    ]
+    UNURAN_SOURCE_DIRS = [UNURAN_DIR / dir_ for dir_ in UNURAN_DIRS[2:]]
+
+    sources = _get_sources(UNURAN_SOURCE_DIRS)
+
+    config.add_library(
```

Comment:
Here's where you can try combining this `add_library` call and the `add_extension` call below into a single shared library:
```python
   ext = config.add_extension(
        "unuran",
        sources=["unuran_wrapper.c"] + sources,
        libraries=[],
        include_dirs=UNURAN_DIRS
        + [
            os.path.join(
                os.path.dirname(__file__), "..", "..", "_lib", "src"
            )
        ]
        + [os.path.join(os.path.dirname(__file__), "unuran_callback.h")],
        language="c",
        define_macros=DEFINE_MACROS,
    )
    ext.pre_build_hook = unuran_pre_build_hook
```

Reply:
Done.